Cargo.toml

```diff
@@ -77,6 +77,7 @@ sqlparser = { version = "0.43.0", features = ["visitor"] }
 tempfile = "3"
 thiserror = "1.0.44"
 tokio = { version = "1.36", features = ["macros", "rt", "sync"] }
+unescaper = "0.1.4"
```

Comment:
I am worried about https://crates.io/crates/unescaper -- it is maintained by a single person and the homepage doesn't work https://hack.ink/unescaper

Its license is also somewhat questionable as it appears to be based on a previously licensed work but doesn't include the appropriate license (e.g.  https://github.com/saghm/unescape-rs/blob/master/LICENSE but the copyright isn't preserved)

Reply:
That would make sense to me (probably the code belong eventually upstream in sqlparser-rs but we could start with it in DataFusion)

Reply:
I have preliminarily implemented unescape function, please review it.

Reply:
> That would make sense to me (probably the code belong eventually upstream in sqlparser-rs but we could start with it in DataFusion)

I think you're right. sqlparser-rs does do some unescape work, but it's not perfect, and we need to improve it.

https://github.com/sqlparser-rs/sqlparser-rs/blob/1a07c5d67c13c0b635cfe0b3e6b13b5be6cae275/src/tokenizer.rs#L1197-L1258